### PR TITLE
Refactor cmake

### LINKED ISSUE #920: Refactor cmake
Currently cmake contains a huge list of all the source files and compiling all of them into one clio library. It would be better to make each directory in `src` as a library with it's own cmake file.

### PR DESCRIPTION
**Pull request by kuznetsss**
Fixes #920.

### CODE REVIEW

File: docker/develop/run [hunk 1]
```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+
+set +x
```
Comment (godexsoft) [inline docker/develop/run]: Do we need this still? Personally I prefer scripts to be silent other than logging through `echo`s.

File: src/util/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+         libbacktrace::libbacktrace
+         fmt::fmt
+         openssl::openssl
+         xrpl::libxrpl
+         Threads::Threads
+         clio_options
+)
+
+if (is_gcc)
+  # FIXME: needed on gcc for now For some reason cmake doesn't propagate the compile definitions from clio_options so we
```
Comment (godexsoft) [inline src/util/CMakeLists.txt]: Please split the comment into two lines "now\nFor some".

**Review by godexsoft — DISMISSED** [review]
This is brilliant. Looks so much more modern 🥇 
Leaving just a couple questions/suggestions 👍

File: docker/develop/run (hunk 1 above)
Comment (kuznetsss) [reply docker/develop/run]: Sorry, left from debug

File: src/util/CMakeLists.txt (hunk 2 above)
Comment (kuznetsss) [reply src/util/CMakeLists.txt]: Hm, it was on different lines until cmake-format merged it 🤔

Comment (godexsoft) [reply src/util/CMakeLists.txt]: Perhaps it wants a `.` at the end of the sentence? :rage2:

**Review by godexsoft — APPROVED** [review]
🚀 :rage1:

**Pull request merged**